### PR TITLE
T15574

### PR DESCRIPTION
**Pull request by smspillaz**
Add support for ESLint during the build. This requires `eslint` to be present in the `PATH`. Use `EOS_ESLINT_RULES` to generate the targets for ESLint support. Add `run_eslint.eslint` to `TESTS` and add `.eslint` to `TESTS_EXTENSIONS` to run ESLint during `make check`.

Fixes numerous errors picked up by ESLint.

https://phabricator.endlessm.com/T15574

### CODE REVIEW

**Comment by smspillaz** [comment]
Rebased and updated

File: tests/js/testTimestamps.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-env jasmine */
```
Comment (cosimoc) [inline tests/js/testTimestamps.js]: Can you avoid C-style comment for this?

File: src/actor.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ const Cairo = imports.cairo;
 const GdkPixbuf = imports.gi.GdkPixbuf;
 const Gio = imports.gi.Gio;
 const GObject = imports.gi.GObject;
+const Gdk = imports.gi.Gdk;
```
Comment (cosimoc) [inline src/actor.js]: Alphabetical order

File: tests/js/testTimestamps.js (hunk 1 above)
Comment (smspillaz) [inline tests/js/testTimestamps.js]: I'm afraid not :( ESLint seems to only detect it if it is in a C style comment (tested again now).

File: src/actor.js (hunk 2 above)
Comment (smspillaz) [inline src/actor.js]: Done

**Pull request merged**